devtools/parse_pcap_klap.py

```diff
@@ -0,0 +1,300 @@
+        self.update_encryption_session()
+
+    @property
+    def local_seed(self) -> bytes:
+        """Get the local seed."""
+        return self._local_seed
+
+    @local_seed.setter
+    def local_seed(self, value: bytes):
+        if not isinstance(value, bytes):
+            raise ValueError("local_seed must be bytes")
+        elif len(value) != 16:
+            raise ValueError("local_seed must be 16 bytes")
+        else:
+            self._local_seed = value
+            self.update_encryption_session()
+
+    @property
+    def remote_auth_hash(self) -> bytes:
+        """Get the remote auth hash."""
+        return self._remote_auth_hash
+
+    @remote_auth_hash.setter
+    def remote_auth_hash(self, value: bytes):
+        print("setting remote_auth_hash")
+        if not isinstance(value, bytes):
+            raise ValueError("remote_auth_hash must be bytes")
+        elif len(value) != 32:
+            raise ValueError("remote_auth_hash must be 32 bytes")
+        else:
+            self._remote_auth_hash = value
+            self.update_encryption_session()
+
+    @property
+    def remote_seed(self) -> bytes:
+        """Get the remote seed."""
+        return self._remote_seed
+
+    @remote_seed.setter
+    def remote_seed(self, value: bytes):
+        print("setting remote_seed")
+        if not isinstance(value, bytes):
+            raise ValueError("remote_seed must be bytes")
+        elif len(value) != 16:
+            raise ValueError("remote_seed must be 16 bytes")
+        else:
+            self._remote_seed = value
+            self.update_encryption_session()
+
+    # This function decrypts the data using the encryption session.
+    def decrypt(self, *args, **kwargs):
+        """Decrypt the data using the encryption session."""
+        if self._session is None:
+            raise ValueError("No session available")
+        return self._session.decrypt(*args, **kwargs)
+
+
+
+# This is a custom error handler that replaces bad characters with '*',
+# in case something goes wrong in decryption.
+# Without this, the decryption could yield an error.
+def bad_chars_replacement(exception):
+    """Replace bad characters with '*'."""
+    return ("*", exception.start + 1)
+
+
+codecs.register_error("bad_chars_replacement", bad_chars_replacement)
+
+
+
+
+
+def main(username, password, device_ip, pcap_file_path, output_json_name=None):
+    """Run the main function."""
+    capture = pyshark.FileCapture(pcap_file_path, display_filter="http")
+
+    # In an effort to keep this code tied into the original code
+    # (so that this can hopefully leverage any future codebase updates inheriently),
+    # some weird initialization is done here
+    myCreds = Credentials(username, password)
+
+    fakeConnection = DeviceConnectionParameters(
+        DeviceFamily.SmartTapoBulb, DeviceEncryptionType.Klap
+    )
+    fakeDevice = DeviceConfig(
+        device_ip, connection_type=fakeConnection, credentials=myCreds
+    )
```

Comment:
```suggestion
    fake_connection = DeviceConnectionParameters(
        DeviceFamily.SmartTapoBulb, DeviceEncryptionType.Klap
    )
    fake_device = DeviceConfig(
        device_ip, connection_type=fakeConnection, credentials=myCreds
    )
```
Use camel_case for variable naming, CamelCase is only reserved for class names & enum members.